### PR TITLE
Text Style: Reset font weight to regular on font family change

### LINKED ISSUE #1899: Changing font family should result in 400 for font weight
## Bug Description

Sometimes when changing the font family, "multiple" is shown as the font weight even if multiple font weights haven't been defined within the text.

It might be that the new font family doesn't have the font weight assigned and that causes it but this needs verifying.

## Expected Behaviour

- No input field for a single selected element should ever have a "multiple" value.

Display the closest available font weight?
TBD by UX

Note: whatever is chosen should be displayed in the design panel and also for the text itself, too -- meaning applying it to the element content.

## Steps to Reproduce

1. Add text element
2. Assign font family Open Sans Condendsed, font weight Light
3. Now change the font family to Playball
4. See "multiple" as the font weight.

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by zachhale**
## Summary

Bring back fix for #1899 and fix lingering bug. The original fix (#4900) implemented an almost complete fix, but left one bug which reverted any content changes in the currently editing text element when the font family was changed.

This PR re-applies the original fix and resolves the bug with content getting reverted when family changes.

Original fix PR: #4900
Revert PR: #5045 

## Relevant Technical Choices

This was a tricky one because the editor needs to get destroyed so that we can (1) set the font weight to the whole element and (2) make sure any content changes are saved first.

There's no way to have the appropriate content to clean up until after the editor closes, so this PR introduces a "queued push" concept that waits for `selectedElements` to update before resetting the font weight on the element.

## Testing Instructions

1. Add text element
2. Type Hey. Hit enter then type more text again
3. Change the font weight on part of the text
4. Highlight any part of the text or ensure the cursor is in the editor
4. Go to font picker and select any font family
5. Editor should go away, font family should change, and all font weights should reset to be one single font weight

---

Fixes #1899

### CODE REVIEW

File: assets/src/edit-story/components/panels/textStyle/useRichTextFormatting.js [hunk 1]
```diff
@@ -98,14 +103,32 @@ function useRichTextFormatting(selectedElements, pushUpdate) {
 
   const push = useCallback(
     (updater, ...args) =>
-      pushUpdate(
-        ({ content }) => ({ content: updater(content, ...args) }),
-        true
-      ),
+      pushUpdate(({ content }) => {
+        return { content: updater(content, ...args) };
+      }, true),
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/textStyle/useRichTextFormatting.js]: nit: this doesn't need the brackets, can just keep it how it was I think?

File: assets/src/edit-story/components/panels/textStyle/useRichTextFormatting.js [hunk 2]
```diff
@@ -98,14 +103,32 @@ function useRichTextFormatting(selectedElements, pushUpdate) {
...
+  const queuedPushRef = useRef(null);
+  const queuePush = useCallback(
+    (...args) => {
+      queuedPushRef.current = args;
+    },
+    [queuedPushRef]
+  );
+
+  // when selected elements update, run any queued pushes
+  useEffect(() => {
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/textStyle/useRichTextFormatting.js:122]: 👏 👏 👏

**Review by BrittanyIRL — APPROVED** [review]
This makes sense! Well done. 

I think added a karma test for this edge case would be useful. It works great!

**Comment by zachhale** [comment]
OPE, looks like those tests were moved to a different file while I was working on this. Brought them back!

**Review by carloskelly13 — APPROVED** [review]
